lib/core/client.d.ts

```diff
@@ -0,0 +1,2 @@
+import start from './dist/client/index';
+export = start;
```

Comment:
This caused all the apps to go from `import { start } from '@storybook/core/client'` to `import client from `@storybook/core/client`. 

However, the actual implementation of `core/client.js`, `core/src/index.js`, and `core/src/preview/index.ts` matches this, as `core/src/client/preview/index.ts` exports an _object_ as `default` instead of exporting multiple functions. Therefore to match that, has to be default-imported in typescript.

`core/src/client/preview/index.ts` could be changed to a more correct module, but I'm not sure if there was a reason it was done the way it is right now.

Reply:
I'm not sure to understand why `import { start } from '@storybook/core/client'` need to become `import client from '@storybook/core/client'`. I added locally this `lib/core/client.d.ts` and has no error with `import { start } from '@storybook/core/client'`.

Should `start` not be `client` instead as `start` is only a property of the exported object?

Reply:
That's strange that it worked fine for you @gaetanmaisse - typescript was definitely erroring for me when I kept it as `import { client }`.

You're right that it _works fine_ once it compiles down to CommonJS, as it's basically `const { start } = require('@storybook/client-api')` either way, but it's not valid in ES modules or typescript to import a single property of a default-exported object. Babel fudges this restriction a bit, when interfacing with CommonJS code.

For correct ES modules, we would need to have:

 `core/src/client/preview/index.ts`
```typescript
export { ClientApi, StoryStore, ConfigApi } from '@storybook/client-api';
export { toId } from '@storybook/csf';
export { default as start } from './start';
```

`core/src/client/index.ts`
```typescript
export * from './preview';
export * from './preview/types';
```

`core/client.d.ts`
```typescript
export * from './dist/client';
```

Which is definitely simpler, but I didn't know if there was a reason for the current way, so I didn't touch that.

Reply:
:+1: thanks for the clarification, I agree with you it will be way simpler/clearer with the solution in your comment. I had the same question about the reason of this kind of export in other parts of Storybook but I never had a strong answer, so it's surely a code style legacy.

Reply:
I'll go ahead and switch to the new style and hope nothing breaks :smile:

Reply:
This is really bizzare, `yarn build core` works, `yarn build react` works, `yarn build core api react vue` works, but `yarn build --all` fails when building `@storybook/react`. Is there anything funky going on when I run `--all` @gaetanmaisse?

Reply:
Hmm I would say no, but maybe 🙈 😂 Is it working with `yarn bootstrap --core`?